src/pages/Schedule.js

```diff
@@ -1,49 +1,170 @@
 import React, { Component } from 'react';
+import ScheduledMatch from '../components/ScheduledMatch';
 import provider from "../dataProvider";
 import Utils from "../utils";
 
 export default class Schedule extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      dates: []
+      dates: [],
+      teams: [],
+      scheduledMatches: [],
+      filteredMatches: [],
+      dateOptionValue: '',
+      teamOptionValue: 'none'
     };
 
     this.fetchListOfPlayingDates();
+    this.fetchListOfScheduledMatches();
+    this.fetchListOfTeams();
+
   }
+  /** Data from Promise  */
 
   fetchListOfPlayingDates() {
     provider.getListOfPlayingDates().then(data => {
-      this.setState({ dates: data.matchDays });
+      const currentDate = new Date('July 2, 2018 23:15:30'); // Current date is chosen for testing purpose for now
+      const nearestDate = data.matchDays.find(el => {
+        const kslDate = new Date(el.date);
+        return kslDate.getTime() > currentDate.getTime()
+      })
+
+      this.setState({
+        dates: data.matchDays,
+        dateOptionValue: nearestDate.date
+      });
     })
   }
 
-  render() {
+  fetchListOfTeams() {
+    provider.getMenuTeamsList().then(data => {
+      this.setState({
+        teams: data.teams,
+      });
+    })
+  }
 
+  fetchListOfScheduledMatches() {
+    provider.getListOfScheduledMatches().then(data => {
+      this.setState({
+        scheduledMatches: data.matches,
+        filteredMatches: data.matches.filter(el => { return this.state.dateOptionValue.toString() === el.date.date.toString() })
+      })
+    })
+  }
+  /************************************************************************************************************************************ */
+
+  /** Event functions - Date Selection and Team Selection */
+  handleOnDateChange = (e) => {
+    const pickedDate = e.target.value
+    const filteredMatches = this.state.scheduledMatches.filter(el => {
+      console.log(" dateOptionValue ETARGET " + e.target.value.toString())
```

Comment:
![image](https://user-images.githubusercontent.com/3020042/58151617-743c5780-7c6a-11e9-9a8d-d2277b74fd40.png)
When I change date in filters, this message is logged 30 times, that means your handle function runs 30 times. Theres something wrong with it because it should run once per every user change

Reply:
okay, I got it now, the console.log is inside filter function, so its okay :)